runtime/src/unchecked_extrinsic.rs

```diff
@@ -0,0 +1,437 @@
+	}
+}
+use crate::signature::CustomSignature;
+use frame_support::weights::{DispatchInfo, GetDispatchInfo};
+use polkadex_primitives::Signature;
+use sp_core::crypto::AccountId32;
+use sp_runtime::traits::Verify;
+
+impl<Address, Call, Extra> GetDispatchInfo for UncheckedExtrinsic<Address, Call, Extra>
+where
+	Call: GetDispatchInfo,
+	Extra: SignedExtension,
+{
+	fn get_dispatch_info(&self) -> DispatchInfo {
+		self.function.get_dispatch_info()
+	}
+}
+
+/// Manual [`TypeInfo`] implementation because of custom encoding. The data is a valid encoded
+/// `Vec<u8>`, but requires some logic to extract the signature and payload.
+///
+/// See [`UncheckedExtrinsic::encode`] and [`UncheckedExtrinsic::decode`].
+impl<Address, Call, Extra> TypeInfo for UncheckedExtrinsic<Address, Call, Extra>
+where
+	Address: StaticTypeInfo,
+	Call: StaticTypeInfo,
+	Extra: SignedExtension + StaticTypeInfo,
+{
+	type Identity = UncheckedExtrinsic<Address, Call, Extra>;
+
+	fn type_info() -> Type {
+		Type::builder()
+			.path(Path::new("UncheckedExtrinsic", module_path!()))
+			// Include the type parameter types, even though they are not used directly in any of
+			// the described fields. These type definitions can be used by downstream consumers
+			// to help construct the custom decoding from the opaque bytes (see below).
+			.type_params(vec![
+				TypeParameter::new("Address", Some(meta_type::<Address>())),
+				TypeParameter::new("Call", Some(meta_type::<Call>())),
+				TypeParameter::new("Signature", Some(meta_type::<Signature>())),
+				TypeParameter::new("Extra", Some(meta_type::<Extra>())),
+			])
+			.docs(&["UncheckedExtrinsic raw bytes, requires custom decoding routine"])
+			// Because of the custom encoding, we can only accurately describe the encoding as an
+			// opaque `Vec<u8>`. Downstream consumers will need to manually implement the codec to
+			// encode/decode the `signature` and `function` fields.
+			.composite(Fields::unnamed().field(|f| f.ty::<Vec<u8>>()))
+	}
+}
+
+#[cfg(feature = "std")]
+impl<Address, Call, Extra> parity_util_mem::MallocSizeOf
+	for UncheckedExtrinsic<Address, Call, Extra>
+where
+	Extra: SignedExtension,
+{
+	fn size_of(&self, _ops: &mut parity_util_mem::MallocSizeOfOps) -> usize {
+		// Instantiated only in runtime.
+		0
+	}
+}
+
+impl<Address, Call, Extra: SignedExtension> UncheckedExtrinsic<Address, Call, Extra> {
+	/// New instance of a signed extrinsic aka "transaction".
+	pub fn new_signed(function: Call, signed: Address, signature: Signature, extra: Extra) -> Self {
+		Self { signature: Some((signed, signature, extra)), function }
+	}
+
+	/// New instance of an unsigned extrinsic aka "inherent".
+	pub fn new_unsigned(function: Call) -> Self {
+		Self { signature: None, function }
+	}
+}
+
+impl<Address, Call, Extra: SignedExtension> Extrinsic for UncheckedExtrinsic<Address, Call, Extra> {
+	type Call = Call;
+
+	type SignaturePayload = (Address, Signature, Extra);
+
+	fn is_signed(&self) -> Option<bool> {
```

Comment:
This is antipattern. Can we use enum for return type here?

Reply:
That trait is controlled by substrate.